File: src/main/java/uknowlab/designpattern/factory/NYPizzaStore.java
package uknowlab.designpattern.factory;

import uknowlab.designpattern.factory.pizza.NYCheesePizza;
import uknowlab.designpattern.factory.pizza.NYPepperoniPizza;
import uknowlab.designpattern.factory.pizza.NYVeggiePizza;
import uknowlab.designpattern.factory.pizza.Pizza;

public class NYPizzaStore extends PizzaStore {

    @Override
    protected Pizza createPizza(String type) {
        return switch (type) {
            case "cheese" -> new NYCheesePizza();
            case "pepperoni" -> new NYPepperoniPizza();
            case "veggie" -> new NYVeggiePizza();
            default -> null;
        };
    }
}

File: src/main/java/uknowlab/designpattern4kotlin/factory/NYPizzaStore.kt
package uknowlab.designpattern4kotlin.factory

import uknowlab.designpattern4kotlin.factory.pizza.NYCheesePizza
import uknowlab.designpattern4kotlin.factory.pizza.NYPepperoniPizza
import uknowlab.designpattern4kotlin.factory.pizza.NYVeggiePizza
import uknowlab.designpattern4kotlin.factory.pizza.Pizza


class NYPizzaStore : PizzaStore() {
    override fun createPizza(type: String?): Pizza {
        return when (type) {
            "cheese" -> NYCheesePizza()
            "pepperoni" -> NYPepperoniPizza()
            "veggie" -> NYVeggiePizza()
            else -> NYCheesePizza()
        }
    }
}
